code/src/app/api/carInfo/route.ts

```diff
@@ -217,144 +166,105 @@ export async function GET(req: Request) {
-    manufacturer,
-    type,
-    year,
-    license,
-    mileage,
-    model,
-    color,
-    seats,
-    condition,
-    mileageLastOilChange,
-    mileageLastTireChange,
-  } = requestBody;
-
-  let {
-    dateNextOilChange,
-    dateNextTireChange,
-    oilOverdue,
-    tireOverdue,
-    maintenanceRequired,
-  } = requestBody;
 
+export async function PUT(req: Request) {
   try {
-    // Get saved car object, validate car exist
-    const getRequest = await GET(req);
-    const { message, carObject } = await getRequest.json();
-    if (!carObject) {
-      return NextResponse.json({ message }, { status: getRequest.status });
-    }
+    const data = await req.json();
 
-    // Check existing license being registered with another car
-    const existingLicense = await car.findOne({
-      // eslint-disable-next-line no-underscore-dangle
-      $and: [{ license }, { _id: { $ne: carObject._id } }],
-    });
+    const entries = Object.entries(data);
+    const nonEmptyOrNull = entries.filter(
+      ([, val]) => val !== '' && val !== null,
+    );
+    const filteredData = Object.fromEntries(nonEmptyOrNull);
 
-    if (existingLicense) {
+    if (!filteredData.carId) {
       return NextResponse.json(
-        { message: 'License number already registered.' },
+        { message: 'Please provide a carId' },
         { status: 400 },
       );
     }
+    await dbConnect();
 
-    // TODO Assumption: If new Date for last maintenance change, then also new mileage for last maintenance change
+    /* Check if license already exists for another car */
 
-    // Type conversion from request to Date object
-    let dateNextOilChangeDate = new Date('');
-    let dateNextTireChangeDate = new Date('');
+    if (filteredData.license) {
+      const existingCarWithSameLicense = await car.findOne({
+        license: filteredData.license,
+        _id: { $ne: filteredData.carId }, // Exclude the car being updated
+      });
 
-    // Oil Maintenance
-    if (
-      isMaintenanceTypeRequired(
-        mileageLastOilChange,
-        rotationMileageOil,
-        mileage,
-      )
-    ) {
-      dateNextOilChangeDate = new Date(dateNextOilChange);
-      oilOverdue = isMaintenanceOverdue(dateNextOilChangeDate, oilOverdue);
-      dateNextOilChangeDate = dateNeedsMaintenance(dateNextOilChangeDate);
-    } else {
-      oilOverdue = false;
-      dateNextOilChange = '';
+      if (existingCarWithSameLicense) {
+        return NextResponse.json(
+          { message: 'License already exists for another car.' },
+          { status: 401 },
+        );
+      }
     }
 
-    // Tire Maintenance
+    /* Check for new date and mileage for last maintenance change */
+
     if (
-      isMaintenanceTypeRequired(
-        mileageLastTireChange,
-        rotationMileageTire,
-        mileage,
-      )
+      (filteredData.dateNextOilChange && !filteredData.mileageLastOilChange) ||
+      (filteredData.dateNextTireChange && !filteredData.mileageLastTireChange)
     ) {
```

Comment:
'If a new date for last maintenance change is provided, then a new mileage for the last maintenance change should also be provided.'
In order to make sure both mileage and date been provide.